source/extensions/dynamic_modules/sdk/rust/README.md

```diff
@@ -0,0 +1,7 @@
+# Envoy Dynamic Modules Rust SDK
+
+This directory contains the Rust SDK for the Dynamic Modules feature. The SDK passes the same set of tests and is guaranteed to provide the same functionality as the other SDKs. This directory is organized in the way that it can be used as a standalone Rust crate. The SDK is basically the high-level abstraction layer for the Dynamic Modules ABI defined in the [abi.h](../../abi.h).
+
+Currently, the ABI binding ([src/abi.rs](./src/abi.rs)) is manually generated by [`bindgen`](https://github.com/rust-lang/rust-bindgen) for the ABI header. Ideally, we should be able to do the bindgen in the build.rs file. 
+
+TODO(@mathetake): figure out how to properly setup the bindgen in build.rs with rules_rust. The most recommended way is to use [crate_universe](https://bazelbuild.github.io/rules_rust/crate_universe.html#setup) and use bindgen as a dev-dependency. However, it seems that crate_universe tries to use the underlying gcc system linker which we cannot assume always available.
```

Comment:
note: i tried **so hard** (like spent the entire two days) to get the crate_universe working in our envoy build settings but no luck. basically it is officially recommended way to deal with cargo dependencies but it seems not that compatible with our settings. So I had to choose to do the manual bindgen for now - which of course not ideal so  I left this TODO.

Reply:
We use crate universe here with bazel: https://github.com/bitdriftlabs/capture-sdk. Not sure if there is anything to be learned from that.

Reply:
I identified the root cause: rules_rust repo declared in Envoy is a bit old and patched version (used by the comptime option Wasmtime dependency and tests for Wasm extensions. They will never be able to use the vanilla rules_rust), hence `crate_universe_dependencies` requires `bootstrap = True` argument due to the reason descrbied in the crate_universe doc:

> Note that if the current version of rules_rust is not a release artifact, you may need to set additional flags such as [bootstrap = True](https://bazelbuild.github.io/rules_rust/crate_universe.html#crate_universe_dependencies-bootstrap) on the crate_universe_dependencies call above or [crates_repository::generator_urls](https://bazelbuild.github.io/rules_rust/crate_universe.html#crates_repository-generator_urls) in uses of crates_repository.

This results in rebuilding cargo-bazel which in turns tries to use the system gcc linker. Since we cannot assume (at least for the current build container and CI settings) gcc exists in the system and set in the PATH, that results in the build failure. 

I think we have a few options here:
1. Nukes Wasmtime and Proxy-Wasm Rust SDK dependency in Envoy Wasm sources/tests.
    * For Wasmtime, I _believe_ nobody_ uses it because in anyway it's build time option. If ever there's a user, they can build Envoy by themselves. And for Proxy-Wasm Rust SDK, it's only used in tests, so easy to justify. 
2. Declare the another rules_rust with the non-patched latest version as `rules_rust_2` or whatever different repo name.
3. Install gcc in the build containers and CI env - but this will break the existing dev environment - we need to inform everyone to install gcc which is obviously not good. Also, this will break for Windows for sure.

what do you think? I think 2. is the easiest one but to be honest, that could be the beginning of a mess. If it's ok, I would go for 1., but not sure either.

Reply:
see https://github.com/envoyproxy/envoy/pull/35323 for discussion of the deps clusterfu

i think the solution is a fix in rules_rust (https://github.com/bazelbuild/rules_rust/issues/2665)

cc @martijneken

Reply:
ok cool, more complex than I expected - I think for now we should go with as-is while waiting for the @phlax's PR got merged. I will migrate here to use crate_universe once it's done

Reply:
well i got an idea

Reply:
ok no luck - i will revert the commits trying to use crate_universe

Reply:
heads up - rules_rust is about to be updated